src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/warmup/ServicesWarmup.kt

```diff
@@ -0,0 +1,76 @@
+package it.pagopa.ecommerce.eventdispatcher.warmup
+
+import it.pagopa.ecommerce.eventdispatcher.services.InboundChannelAdapterLifecycleHandlerService
+import it.pagopa.ecommerce.eventdispatcher.warmup.annotations.WarmupFunction
+import kotlin.reflect.full.declaredMemberFunctions
+import kotlin.reflect.full.hasAnnotation
+import kotlin.system.measureTimeMillis
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.getBeansWithAnnotation
+import org.springframework.context.ApplicationListener
+import org.springframework.context.event.ContextRefreshedEvent
+import org.springframework.stereotype.Component
+import org.springframework.stereotype.Service
+import org.springframework.util.ClassUtils
+
+@Component
+class ServicesWarmup(
+  @Autowired
+  private val inboundChannelAdapterLifecycleHandlerService:
+    InboundChannelAdapterLifecycleHandlerService
+) : ApplicationListener<ContextRefreshedEvent> {
+
+  private val logger = LoggerFactory.getLogger(this.javaClass)
+
+  override fun onApplicationEvent(event: ContextRefreshedEvent) {
+    val restservices =
+      event.applicationContext
+        .getBeansWithAnnotation(Service::class.java)
+        .map { it.value }
+        .filter { service ->
+          service::class.java.methods.any { method -> method.name == "messageReceiver" }
+        }
+    logger.info("Found services: [{}]", restservices.size)
+
+    try {
+      restservices.forEach(this::warmUpservice)
+    } catch (e: Exception) {
+      logger.error("Exception during service warm-up", e)
+    } finally {
+      inboundChannelAdapterLifecycleHandlerService.invokeCommandForAllEndpoints("start")
+    }
+  }
+
+  private fun warmUpservice(serviceToWarmUpInstance: Any) {
```

Comment:
```suggestion
  private fun warmUpService(serviceToWarmUpInstance: Any) {
```

code style: use camel case for method declaration

Reply:
done here 13d800efe1c69851d247fe1e49bee5116ee46c50